### PR TITLE
Update README to reflect Elixir 1.4.x Mixfile convention

### PR DESCRIPTION
**Pull request by amorphid**

### CODE REVIEW

**Comment by optikfluffel** [comment]
@amorphid could you also bump the version to `0.11` in README while you're on it? ^^

**Comment by amorphid** [comment]
to be clear, you mean change from `0.11.0` to `0.11`?  AKA dropping the last 0?

**Comment by amorphid** [comment]
sorry, I misread.  I am making change now.  I thought you meant `mix.exs`, which confused me greatly :)

**Comment by amorphid** [comment]
@optikfluffel Done.

**Comment by edgurgel** [comment]
So if I understood well, Elixir 1.4 will already find which applications will be started so there's no need to define anything at all, right?

So just `< 1.4` should put something inside mix.exs ?

**Comment by amorphid** [comment]
@edgurgel Good question.  Would it make sense to upgrade the Elixir version to 1.4, and recommend that people using Elixir 1.3 or below to stick w/ v0.11.0?

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ First, add HTTPoison to your `mix.exs` dependencies:
 
 ```elixir
 def deps do
-  [{:httpoison, "~> 0.10.0"}]
+  [{:httpoison, "~> 0.11.0"}]
```
Comment (bbtb1982) [inline README.md]: I would just put [{:httpoison, "~> x.x.x"}]  and add a sentence to look to the most recent stable version listed in releases.

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ application dependency:
 
 ```elixir
 def application do
-  [applications: [:httpoison]]
+  [extra_applications: [:httpoison]]
```
Comment (bbtb1982) [inline README.md]: As for the comment about 1.4 and not needing to include the app. It appears to not hurt to list the packages and as such I would also just leave it as it is more explicit and makes it easier to know what is actually being used.

Comment (m4ttsch) [inline README.md]: I disagree that this should be explicit. The new feature that shipped with Elixir 1.4 is application **inference**, which implies that you no longer need to be explicit. 

From [v1.4 Release Notes](https://elixir-lang.org/blog/2017/01/05/elixir-v1-4-0-released/):

"Mix will automatically build your application list based on your dependencies. Developers now only need to specify which applications shipped as part of Erlang or Elixir that they require, such as :logger."

I would just make a note to only add `:httpoison` to your application list if your elixir version is 1.3 or lower.

Comment (edgurgel) [inline README.md]: I agree @m4ttsch !

**Comment by edgurgel** [comment]
Done: https://github.com/edgurgel/httpoison/commit/11b35f57a508338c6b9926485498d6d73c13090c

**Comment by amorphid** [comment]
me, too.  closing.

**Pull request closed without merging**